### PR TITLE
add data_source_huaweicloud_identity_custom_role

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
Test result:
```bash
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccHuaweiCloudIdentityCustomRoleDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccHuaweiCloudIdentityCustomRoleDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccHuaweiCloudIdentityCustomRoleDataSource_basic
=== PAUSE TestAccHuaweiCloudIdentityCustomRoleDataSource_basic
=== CONT  TestAccHuaweiCloudIdentityCustomRoleDataSource_basic
--- PASS: TestAccHuaweiCloudIdentityCustomRoleDataSource_basic (8.58s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       8.624s
```

### CODE REVIEW

File: huaweicloud/config.go [hunk 1]
```diff
@@ -76,6 +76,8 @@ func (c *Config) LoadAndValidate() error {
 		return fmt.Errorf("max_retries should be a positive value")
 	}
 
+	log.Printf("xxxxx username: %s, password: %s, AK: %s, SK: %s", c.Username, c.Password, c.AccessKey, c.SecretKey)
+
```
Comment (ShiChangkuo) [inline huaweicloud/config.go]: this log includes sensitive arguments, shoud used for debug only.

File: huaweicloud/data_source_huaweicloud_identity_custom_role.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+			"references": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```
Comment (ShiChangkuo) [inline huaweicloud/data_source_huaweicloud_identity_custom_role.go]: `id` is the system attribute which  do not define manually.

File: huaweicloud/data_source_huaweicloud_identity_custom_role.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	if len(allRoles) < 1 {
+		return fmt.Errorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+
+	if len(allRoles) > 1 {
+		log.Printf("[DEBUG] Multiple results found: %#v", allRoles)
+		return fmt.Errorf("Your query returned more than one result. Please try a more " +
+			"specific search criteria, or set `most_recent` attribute to true.")
```
Comment (ShiChangkuo) [inline huaweicloud/data_source_huaweicloud_identity_custom_role.go]: no agrument named `most_recent`, please have a look

**Comment by ShiChangkuo** [comment]
suggest to define more attributes, so we can get more details from the data source.

File: huaweicloud/config.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/config.go]: fixed

File: huaweicloud/data_source_huaweicloud_identity_custom_role.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/data_source_huaweicloud_identity_custom_role.go]: fixed

File: huaweicloud/data_source_huaweicloud_identity_custom_role.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/data_source_huaweicloud_identity_custom_role.go]: It's better if id can be used as a query condition. So I turned the id into an optional parameter.

File: huaweicloud/data_source_huaweicloud_identity_custom_role_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+		return nil
+	}
+}
+
+func testAccHuaweiCloudIdentityCustomRoleDataSource_basic(rName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_identity_role" test {
+	name = "%s"
```
Comment (ShiChangkuo) [inline huaweicloud/data_source_huaweicloud_identity_custom_role_test.go]: please indent two spaces for each nesting level.

Comment (Jason-Zhang9309) [reply huaweicloud/data_source_huaweicloud_identity_custom_role_test.go]: sure. Sorry for not paying attention.

**Pull request merged**